### PR TITLE
feat: peer discovery and routing section

### LINKED ISSUE #250: Create and populate discovery & routing section
## Done Criteria 
- [x] Initiate overview doc on peer discovery
- [x] Initiate protocol docs for:
   
   - [x] Rendezvous
   - [x] mDNS
   - [ ] GossipSub
   - [ ] Kad-DHT

## Why Important
- Provides material on how peers discover and announce services to other peers.
- Documents protocols not yet documented.

## Notes
Populating the docs to be tracked in other issues.

Would follow a structure similar to the other sections, such as:
- What is Peer Discovery
- Rendezvous
- mDNS
- GossipSub
- Kad-DHT

### PR DESCRIPTION
**Pull request by DannyS03**
## Docs this PR introduces that were already approved
* https://github.com/libp2p/docs/pull/260
* https://github.com/libp2p/docs/pull/290
* https://github.com/libp2p/docs/pull/259

## Context
- Creates a new section for peer discovery and routing.
- Transforms the initial intent with the discovery section to capture "peer discovery and routing"
- resolves #250, #13

### CODE REVIEW

**Comment by DannyS03** [comment]
Link fixes and potential conflicts will be addressed in this PR once all associated material has been added.

File: content/concepts/discovery-routing/overview.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+These methods include, but are not limited to:
+
+- [Rendezvous](rendezvous.md): a protocol that allows peers to exchange peer multiaddresses
+  in a secure and private manner.
+- [mDNS](mdns.md): a multicast Domain Name System (DNS) protocol that allows peers to
+  discover other peers on the local network.
+- [Publish/Subscribe](pubsub.md): a protocol that allows peers to subscribe to specific
+  topics and receive updates from other peers that are publishing information on those
+  topics.
```
Comment (p-shahi) [inline content/concepts/discovery-routing/overview.md:55]:
```suggestion
```

File: content/concepts/discovery-routing/pubsub.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+title: "PubSub"
+description: "."
+weight: 225
+---
```
Comment (p-shahi) [inline content/concepts/discovery-routing/pubsub.md:5]: I would move PubSub out of the discovery and routing section and instead place it a section like messaging. Pubsub can be employed for routing/discovery but it's more for message dissemination imo

Comment (DannyS03) [reply content/concepts/discovery-routing/pubsub.md:5]: I intended to remove this (and didn't create another doc as a result), thanks!

**Comment by mxinden** [comment]
@p-shahi is this still planned to merge eventually? If so, I would push a couple of changes to the mdns document.

**Comment by p-shahi** [comment]
@mxinden please push your changes. I will merge by end of week

File: content/concepts/discovery-routing/mdns.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (mxinden) [inline content/concepts/discovery-routing/mdns.md:1]: Is this a duplicate file to the one above? `mdns.md` and `mDNS.md`

**Review by mxinden — APPROVED** [review]
Good to merge from my end.

**Pull request merged**